### PR TITLE
Add word-break to sidebar-item-link

### LINKED ISSUE #22953: Support long projects name in Issue sidebar
### Description

I didn't check the others in sidebar. It is necessary to check all of them.
image

### Gitea Version

latest

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build

### Database

None

### PR DESCRIPTION
**Pull request by yp05327**
Fixes https://github.com/go-gitea/gitea/issues/22953
image

### CODE REVIEW

**Review by delvh — APPROVED** [review]
Yep, that's one approach.
The other would have been to use overflow.
I'm fine with both.

**Review by yardenshoham — APPROVED** [review]
Very good!
image

**Comment by yardenshoham** [comment]
Should we backport to v1.18?

**Review by silverwind — CHANGES_REQUESTED** [review]
~~`overflow-wrap: break-word` should be preferred over `word-break: break-all` because the former results in better breaking where it will first look for natural break opportunities (whitespace) and only break inside a word as last-resort.~~

**Comment by silverwind** [comment]
https://jsfiddle.net/silverwind/jy9v0r4d/ to show the difference between the two.

File: web_src/less/_repository.less [hunk 1]
```diff
@@ -3250,8 +3250,8 @@ td.blob-excerpt {
 }
 
 .sidebar-item-link {
-  display: inline-flex;
   align-items: center;
+  word-break: break-all;
```
Comment (silverwind) [inline web_src/less/_repository.less:3254]:
```suggestion
  overflow-wrap: break-word;
```

**Comment by silverwind** [comment]
Actually, I think `word-break` is fine, reason being that its style of line breaking may be preferred in programming context because that's how editors usually wrap code as well:

image

image

First case acutally looks better to me, being used to how editors break lines.

**Review by silverwind — APPROVED** [review]

**Comment by yp05327** [comment]
@silverwind 
I found that there were some places using them both. I'm not familiar with css, is it possible (or correct?) to use them both?
image
image

**Comment by silverwind** [comment]
Can do such corrections in another PR. Having both `overflow-wrap: break-word` and `word-break: break-all` will probably result in soft-break style word breaks so the `word-break` has no effect and can be removed.

 Whether one or the other is to be used depends on context. If the text to be broken is prose, it should use `overflow-wrap`, otherwise `word-break` is an option in cases where hard breaks are acceptable.

**Comment by yardenshoham** [comment]
Build failure unrelated

**Comment by techknowlogick** [comment]
ping LG-TM bot

**Pull request merged**